### PR TITLE
fs: use dup/dup2 for stdio hide and restore

### PR DESCRIPTION
**Pull request by sreimers**

### CODE REVIEW

File: src/sys/fs.c [hunk 1]
```diff
@@ -211,25 +215,25 @@ int fs_fopen(FILE **fp, const char *file, const char *mode)
...
+ * Hide/Close stdout and stderr output (no THREAD-SAFETY)
  */
 void fs_stdio_hide(void)
 {
-	(void)fclose(stdout);
-	(void)fclose(stderr);
+	dup_stdout = dup(1);
+	dup_stderr = dup(2);
+	(void)close(1);
+	(void)close(2);
```
Comment (alfredh) [inline src/sys/fs.c:225]: can we use `STDOUT_FILENO` and `STDERR_FILENO` instead of 1 and 2 ?

**Comment by alfredh** [comment]
I tested with retest and the "OK" is not printed:

```
$ ./retest -r
retest: libre version 2.9.0 (x86_64/Darwin)
using async polling method 'kqueue'
regular tests:       Alfreds-MBP:retest alfredh$ 
```


is the restore function working ?

**Comment by sreimers** [comment]
Thanks for testing on macOS, will try to debug, on Linux it works fine and better than `freopen`. Needs more testing on Windows too.

**Comment by sreimers** [comment]
ok, retest restore is broken on linux too, so I think we have to fix the usage in retest. Simple usage like here https://github.com/baresip/baresip/blob/main/modules/portaudio/portaudio.c#L306-L308 works fine.

**Comment by alfredh** [comment]
this one is good to go

**Pull request merged**